### PR TITLE
Remove PROFILE_STAGES_API and the dependent code

### LINKED ISSUE #3595: Remove PROFILE_STAGES_API
This feature is no longer maintained and is prone to errors. The output of the profiler is also not very useful anymore.

### PR DESCRIPTION
**Pull request by lubosmj**
closes #3595

### CODE REVIEW

**Comment by lubosmj** [comment]
Consider merging this PR for 3.25.

File: CHANGES/3595.removal [hunk 1]
```diff
@@ -0,0 +1 @@
+Removed the PROFILE_STAGES_API setting along with the dependent code.
```
Comment (ipanova) [inline CHANGES/3595.removal:1]: Removed the PROFILE_STAGES_API in tech preview setting along with the dependent code.

**Review by ipanova — APPROVED** [review]

**Comment by dralley** [comment]
IMO we do not need to wait until 3.25 to remove it, this is a developer only option which developers have not actually used in a long time :)

**Review by dralley — APPROVED** [review]

**Comment by dralley** [comment]
@bmbouter would you be comfortable removing this in 3.24?

**Pull request merged**